cropper/src/main/java/com/canhub/cropper/BitmapUtils.kt

```diff
@@ -0,0 +1,933 @@
+package com.canhub.cropper
+
+import android.content.ContentResolver
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Bitmap.CompressFormat
+import android.graphics.BitmapFactory
+import android.graphics.BitmapRegionDecoder
+import android.graphics.Matrix
+import android.graphics.Rect
+import android.graphics.RectF
+import android.net.Uri
+import android.util.Log
+import android.util.Pair
+import androidx.core.content.FileProvider
+import androidx.exifinterface.media.ExifInterface
+import com.canhub.cropper.CropImageView.RequestSizeOptions
+import com.canhub.cropper.common.CommonValues
+import com.canhub.cropper.common.CommonVersionCheck.isAtLeastQ29
+import java.io.Closeable
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import java.lang.ref.WeakReference
+import javax.microedition.khronos.egl.EGL10
+import javax.microedition.khronos.egl.EGLConfig
+import javax.microedition.khronos.egl.EGLContext
+
+/**
+ * Utility class that deals with operations with an ImageView.
+ */
+internal object BitmapUtils {
+
+    val EMPTY_RECT = Rect()
+    val EMPTY_RECT_F = RectF()
+
+    /**
+     * Reusable rectangle for general internal usage
+     */
+    val RECT = RectF()
+
+    /**
+     * Reusable point for general internal usage
+     */
+    val POINTS = FloatArray(6)
+
+    /**
+     * Reusable point for general internal usage
+     */
+    val POINTS2 = FloatArray(6)
+
+    /**
+     * Used to know the max texture size allowed to be rendered
+     */
+    private var mMaxTextureSize = 0
+
+    /**
+     * used to save bitmaps during state save and restore so not to reload them.
+     */
+    var mStateBitmap: Pair<String, WeakReference<Bitmap>>? = null
+
+    /**
+     * Rotate the given image by reading the Exif value of the image (uri).<br></br>
+     * If no rotation is required the image will not be rotated.<br></br>
+     * New bitmap is created and the old one is recycled.
+     */
+    fun rotateBitmapByExif(bitmap: Bitmap?, context: Context, uri: Uri?): RotateBitmapResult {
+        var ei: ExifInterface? = null
+        try {
+            val `is` = context.contentResolver.openInputStream(uri!!)
+            if (`is` != null) {
+                ei = ExifInterface(`is`)
+                `is`.close()
+            }
+        } catch (ignored: Exception) {
+        }
+        return if (ei != null) rotateBitmapByExif(bitmap, ei) else RotateBitmapResult(bitmap, 0)
+    }
+
+    /**
+     * Rotate the given image by given Exif value.<br></br>
+     * If no rotation is required the image will not be rotated.<br></br>
+     * New bitmap is created and the old one is recycled.
+     */
+    fun rotateBitmapByExif(bitmap: Bitmap?, exif: ExifInterface): RotateBitmapResult {
+        val degrees: Int
```

Comment:
Believe this can be with line 91.

`val degrees: Int = when (orientation) {...}`

Reply:
Got it